src/main/java/Duke.java

```diff
@@ -1,10 +1,203 @@
+
 public class Duke {
+    private static final String LOGO = System.lineSeparator()
+            + "    ___   ______   ________  _______      ___   ____  _____   .--." + System.lineSeparator()
+            + "  .'   `.|_   _ \\ |_   __  ||_   __ \\   .'   `.|_   \\|_   _|.'_\\/_'." + System.lineSeparator()
+            + " /  .-.  \\ | |_) |  | |_ \\_|  | |__) | /  .-.  \\ |   \\ | |  '. /\\,.'" + System.lineSeparator()
+            + " | |   | | |  __'.  |  _| _   |  __ /  | |   | | | |\\ \\| |    \"||\"" + System.lineSeparator()
+            + " \\  `-'  /_| |__) |_| |__/ | _| |  \\ \\_\\  `-'  /_| |_\\   |_    || /\\`" + System.lineSeparator()
+            + "  `.___.'|_______/|________||____| |___|`.___.'|_____|\\____|/\\ ||//\\)" + System.lineSeparator()
+            + "                                                           (/\\\\||/" + System.lineSeparator()
+            + "^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\\||^^^^";
+
+    private static final String HORIZONTAL_BAR = "__________________"
+            + "__________________"
+            + "__________________"
+            + "_______________";
+
+
+    private static final Scanner scanner = new Scanner(System.in);
+    private static ArrayList<Task> tasks = new ArrayList<Task>();
+
+    /**
+     * Print greeting message upon starting the program
+     */
+    public static void displayGreetingMessage() {
+        echo(LOGO + System.lineSeparator() + "  Hello! I'm Oberon"
+                + System.lineSeparator() + "  What can I do for you?");
+    }
+
+    /**
+     * Print farewell message upon exiting the program
+     */
+    public static void displayFarewellMessage() {
+        echo("  Goodbye. Hope to see you again soon!" + System.lineSeparator() + LOGO);
+    }
+
+    /**
+     * Reads in a string and prints it with a horizontal bar above & below it
+     * @param input String to be echo-ed
+     */
+    public static void echo(String input) {
+        System.out.println(HORIZONTAL_BAR + System.lineSeparator() + input
+                + System.lineSeparator() + HORIZONTAL_BAR);
+    }
+
+    /**
+     * Prints current list of tasks
+     */
+    public static void printTasks() {
+        if (tasks.isEmpty()) {
+            echo("  List is empty!");
+        } else {
+            System.out.println(HORIZONTAL_BAR);
+            // Traverse down the ArrayList and prints out each task
+            for (int i = 0; i < tasks.size(); i++) {
+                int currentIndexInOnesIndexing = i + 1;
+                System.out.println("  " + currentIndexInOnesIndexing + ". "
+                        + tasks.get(i).getTaskDescriptionWithStatus());
+            }
+            System.out.println(HORIZONTAL_BAR);
+        }
+    }
+
+    /**
+     * Marks task in array as done and prints echo message depending on validity of task no.
+     */
+    public static void markTaskAsDone(String input) {
+        // Extracting task number as an integer from input
```

Comment:
In line with other comments, perhaps you can phrase it as a command/action instead of using the "ing" form for "Extracting"